src/writer/creator.cpp

```diff
@@ -254,16 +254,21 @@ namespace zim
       // We can now stop the direntHandlers, and get their content
       for(auto& handler:data->m_direntHandlers) {
         handler->stop();
-        auto dirent = handler->getDirent();
-        if (dirent == nullptr) {
+        auto dirents = handler->getDirents();
+        if (dirents.empty()) {
           continue;
         }
-        auto provider = handler->getContentProvider();
-        data->addItemData(dirent, std::move(provider), handler->isCompressible());
-        if (handler == data->mp_titleListingHandler) {
-          // We have to get the offset of the titleList in the cluster before
-          // we close the cluster. Once the cluster is close, the offset information is dropped.
-          data->m_titleListBlobOffset = data->uncompCluster->getBlobOffset(dirent->getBlobNumber());
+        auto providers = handler->getContentProviders();
+        ASSERT(dirents.size(), ==, providers.size());
+        auto provider_it = providers.begin();
+        for(auto& dirent:dirents) {
+          data->addItemData(dirent, std::move(*provider_it), handler->isCompressible());
```

Comment:
This means that for a handler that creates multiple dirents/items all of them must be compressible or not at the same time.

Reply:
Yes. It is ok for now as handler are creating same kind of content.
I've added a comment to explicit this.